packages/core/data-transfer/src/strapi/providers/local-source/assets.ts

```diff
@@ -94,10 +94,9 @@ export const createAssetsStream = (strapi: LoadedStrapi): Duplex => {
           const fileFormatFilepath = isLocalProvider
             ? join(strapi.dirs.static.public, fileFormat.url)
             : fileFormat.url;
-
           const fileFormatStats = await getFileStats(fileFormatFilepath, isLocalProvider);
           const fileFormatStream = getFileStream(fileFormatFilepath, isLocalProvider);
-          const metadata = { ...fileFormat, type: format, mainHash: file.hash };
+          const metadata = { ...fileFormat, type: format, id: file.id };
```

Comment:
Do you think this could be id, with a fallback on mainHash (like we just discussed in slack), so that it doesn't break existing export files? Remote transfers might still have the bug when going from old strapi <-> fixed strapi, but that's fine, we will document and advise that both source and destination need to be on the fixed version for it to actually be fixed.

Reply:
That could be why all the e2e tests are failing, this could be breaking the setup which uses data-transfer to import a test dataset.

Reply:
not anymore